### PR TITLE
Retrieve decklists from NetrunnerDB

### LINKED ISSUE #114: NetrunnerDB-based deck registration UI
We'd like a UI for a player to choose from their private decks in NetrunnerDB when they register for a tournament. We'll cover saving it to the database as a separate issue, so this option should be marked as incomplete or hidden in production.

We can add a flag in the tournament setup to turn on deck registration via NetrunnerDB. This would replace the current inputs for corp and runner ID. The button to register would take you to a separate screen where you'd register your decks. This would display your recent private decks from NetrunnerDB, and let you choose a corp and runner.

The screen should have a link to NetrunnerDB in case you haven't built your deck there yet, or if they haven't copied it to their private decks.

#### Authentication

The NetrunnerDB token might have expired when you open this UI. If that happens, you should be reauthenticated when it's time to display the list of decks.

#### Covered by other issues

It might be nice to be able to choose a deck that's already been published, and for Cobra to remember that it's already published. We probably don't want to handle that, at least not yet.

There's a separate issue to show the list of cards in the deck once you've chosen it, as it will appear to the TO: https://github.com/Project-NISEI/cobra/issues/115

We'll want to detect the identity of the deck to store it against the player for stats and display. This could be done as part of this or in the card display issue.

The option to submit the deck registration should be greyed out, and handled by https://github.com/Project-NISEI/cobra/issues/116.

### PR DESCRIPTION
**Pull request by haradan**
Resolves #114.

Initial UI for choosing a deck to register loaded from the NetrunnerDB API.

Other issues will follow up on this to store the registration (https://github.com/Project-NISEI/cobra/issues/116), display it to the TO (https://github.com/Project-NISEI/cobra/issues/117), etc.

image

Here's how it looks if you haven't created any decks yet:

image

### CODE REVIEW

File: app/views/tournaments/edit.html.slim
Comment (plural) [inline app/views/tournaments/edit.html.slim:1]: I generally like to pull extra things like formatting changes only out into their own PRs - they aren't related to the functional change.

File: config/spring.rb
Comment (plural) [inline config/spring.rb:1]: a few of your files are missing newlines at the end.  Would you please check your editor settings?

**Review by plural — CHANGES_REQUESTED** [review]
this is a great start!

File: app/views/tournaments/edit.html.slim
Comment (haradan) [reply app/views/tournaments/edit.html.slim:1]: I've split this out to PR #128.

File: config/spring.rb
Comment (haradan) [reply config/spring.rb:1]: This file was generated automatically by my IDE when I ran in debug mode. It looks like it only cares about newlines around functions and classes. I've added them manually.

It looks like the other places in this PR are where there wasn't a newline before and there is one now.

**Comment by plural** [comment]
This looks pretty great.  How do you want to feature flag this?

File: app/views/tournaments/_deck_selection.html.slim [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.col-md-6
+  .card
+    .card-header Please select a corp and runner from your decks below
+    ul.list-group.list-group-flush.overflow-auto#nrdb_decks style='height: 23em'
+      - @decks.each do |deck|
+        li.list-group-item id='nrdb_deck_#{deck[:id]}' data-side='#{deck[:side]}' onclick='selectDeck(#{deck[:id]}, "#{deck[:side]}")'
+          p.mb-1= deck[:name]
+          - unless deck[:identity].nil?
```
[the remark this answers is not in the document]
Comment (haradan) [reply app/views/tournaments/_deck_selection.html.slim:15]: I'm not really happy with the pinning behaviour yet. It covers up some of the list so if you select one further down you can't select the top one without selecting another one on the first page first.

**Comment by haradan** [comment]
> This looks pretty great. How do you want to feature flag this?

I've just added Flipper: https://github.com/jnunemaker/flipper

File: app/views/tournaments/_deck_selection.html.slim (hunk 1 above)
[the remark this answers is not in the document]
Comment (haradan) [reply app/views/tournaments/_deck_selection.html.slim:22]: I've made it just always show the selected decks, but I've made it a smaller display.

**Comment by haradan** [comment]
Closing, reopened from a fork: https://github.com/Project-NISEI/cobra/pull/132

**Pull request closed without merging**